src/backend/distributed/executor/multi_client_executor.c

```diff
@@ -247,6 +305,31 @@ MultiClientDisconnect(int32 connectionId)
 }
 
 
+/*
+ * MultiClientReleaseConnection removes a connection from the client
+ * executor pool without disconnecting.
+ *
+ * This allows the connection to be used for other operations in the
+ * same transaction. The connection will still be closed at COMMIT
+ * or ABORT time.
+ */
+void
+MultiClientReleaseConnection(int32 connectionId)
+{
+	MultiConnection *connection = NULL;
+	const int InvalidPollingStatus = -1;
+
+	Assert(connectionId != INVALID_CONNECTION_ID);
+	connection = ClientConnectionArray[connectionId];
+	Assert(connection != NULL);
+
+	UnclaimConnection(connection);
```

Comment:
We could consider closing the connection here if we're not (yet) in a coordinated transaction to keep the current behaviour.

Reply:
I think it would be better to close here when not in a coordinated transaction.